test/local/account_routes.js

```diff
@@ -56,699 +56,453 @@ var makeRoutes = function (options) {
+        t.equal(mockLog.increment.getCall(4).args[0], 'device.update.pushPublicKey')
+      })
+    }, t)
 
-    return new P(function(resolve) {
-      getRoute(accountRoutes, '/account/device')
-        .handler(mockRequest, function(response) {
-          resolve(response)
-        })
-    })
-    .then(
-      function(response) {
+    test('device updates disabled', function (t) {
+      config.deviceUpdatesEnabled = false
+
+      return runTest(route, mockRequest, function () {
         t.fail('should have thrown')
-      },
-      function(err) {
+      })
+      .catch(function (err) {
         t.equal(err.output.statusCode, 503, 'correct status code is returned')
         t.equal(err.errno, error.ERRNO.FEATURE_NOT_ENABLED, 'correct errno is returned')
-      }
-    )
-  }
-)
-
-test(
-  'login event from /account/create includes metrics context data',
-  function (t) {
-    var mockRequest = {
-      app: {
-        acceptLangage: 'en-US'
-      },
-      headers: {
-        'user-agent': 'test-user-agent'
-      },
-      payload: {
-        email: TEST_EMAIL,
-        authPW: crypto.randomBytes(32).toString('hex'),
-        service: 'sync',
-        metricsContext: {
-          entrypoint: 'preferences',
-          utmContent: 'some-content-string'
-        }
-      }
-    }
-    var mockDB = {
-      emailRecord: sinon.spy(function () {
-        return P.reject(new error.unknownAccount())
-      }),
-      createAccount: sinon.spy(function () {
-        return P.resolve({
-          uid: uuid.v4('binary'),
-          email: TEST_EMAIL,
-          emailVerified: false
-        })
       })
-    }
-    // We want to test what's actually written to stdout by the logger.
-    var mockLog = log('ERROR', 'test', {
-      stdout: {
-        on: sinon.spy(),
-        write: sinon.spy()
-      },
-      stderr: {
-        on: sinon.spy(),
-        write: sinon.spy()
-      }
-    })
-    var accountRoutes = makeRoutes({
-      db: mockDB,
-      log: mockLog
-    })
-    return new P(function (resolve) {
-      getRoute(accountRoutes, '/account/create')
-        .handler(mockRequest, function(response) {
-          resolve(response)
-        })
-    })
-    .then(function () {
-      t.equal(mockDB.createAccount.callCount, 1, 'createAccount was called')
-
-      t.equal(mockLog.stdout.write.callCount, 1, 'an sqs event was logged')
-      var eventData = JSON.parse(mockLog.stdout.write.getCall(0).args[0])
-      t.equal(eventData.event, 'login', 'it was a login event')
-      t.equal(eventData.data.service, 'sync', 'it was for sync')
-      t.equal(eventData.data.email, TEST_EMAIL, 'it was for the correct email')
-      t.equal(eventData.data.metricsContext.entrypoint, 'preferences', 'it contained the entrypoint metrics field')
-      t.equal(eventData.data.metricsContext.utm_content, 'some-content-string', 'it contained the utm_content metrics field')
-
-    }).finally(function () {
-      mockLog.close()
```

Comment:
For tests that use a real log object rather than one from `test/mocks`, it must be called at the end of the bounding test in order for the test to terminate cleanly. Other tests don't need it (there's is even a `close` method to call in those cases).